libraries/hotstuff/qc_chain.cpp

```diff
@@ -1054,68 +1055,57 @@ namespace eosio { namespace hotstuff {
-              ("block_id", last_exec_prop->block_id)
-              ("phase_counter", last_exec_prop->phase_counter)
-              ("parent_id", last_exec_prop->parent_id));
+                 ("block_num", last_exec_prop->block_num())("proposal_id", last_exec_prop->proposal_id)("block_id", last_exec_prop->block_id)
+                 ("phase_counter", last_exec_prop->phase_counter)("parent_id", last_exec_prop->parent_id));
 
          fc_tlog(_logger, " *** last_exec_prop ${proposal_id_1} ${phase_counter_1} vs proposal ${proposal_id_2} ${phase_counter_2} ",
-              ("proposal_id_1", last_exec_prop->block_num())
-              ("phase_counter_1", last_exec_prop->phase_counter)
-              ("proposal_id_2", proposal.block_num())
-              ("phase_counter_2", proposal.phase_counter));
+                 ("proposal_id_1", last_exec_prop->block_num())("phase_counter_1", last_exec_prop->phase_counter)
+                 ("proposal_id_2", proposal->block_num())("phase_counter_2", proposal->phase_counter));
       } else {
          fc_tlog(_logger, " === _b_exec proposal is null vs proposal ${proposal_id_2} ${phase_counter_2} ",
-              ("proposal_id_2", proposal.block_num())
-              ("phase_counter_2", proposal.phase_counter));
+                 ("proposal_id_2", proposal->block_num())("phase_counter_2", proposal->phase_counter));
       }
 
-      if (_b_exec == NULL_PROPOSAL_ID)
+      if (_b_exec == NULL_PROPOSAL_ID) {
          exec_height_check = true;
-      else
-         exec_height_check = last_exec_prop->get_height() < proposal.get_height();
-
-      if (exec_height_check){
+      } else {
+         exec_height_check = last_exec_prop->get_height() < proposal->get_height();
+      }
 
-         const hs_proposal_message *p = get_proposal( proposal.parent_id );
+      if (exec_height_check) {
+         const hs_proposal_message* p = get_proposal(proposal->parent_id);
          if (p != nullptr) {
-            //fc_tlog(_logger, " === recursively committing" );
-            commit(*p); //recursively commit all non-committed ancestor blocks sequentially first
+            proposal_stack.push(p); // Push the parent proposal onto the stack for processing
          }
 
-         //Execute commands [...]
-
+         // Execute commands [...]
          fc_dlog(_logger, " === ${id} committed proposal #${block_num} phase ${phase_counter} block_id : ${block_id} proposal_id : ${proposal_id}",
-                        ("id", _id)
-                        ("block_num", proposal.block_num())
-                        ("phase_counter", proposal.phase_counter)
-                        ("block_id", proposal.block_id)
-                        ("proposal_id", proposal.proposal_id));
-      }
-      else {
+                 ("id", _id)("block_num", proposal->block_num())("phase_counter", proposal->phase_counter)("block_id", proposal->block_id)("proposal_id", proposal->proposal_id));
```

Comment:
This log statement will now appear **before** the parent proposals ones, which I think is the wrong order.

Also the `proposal_stack` as it is now is not very useful as we `pop` immediately after `push`, so the size of the stack never exceeds one. We could use a variable instead of a stack. The stack would only be needed if we wanted to preserve the order of the `fc_dlog` statements.